src/integrations/prefect-kubernetes/prefect_kubernetes/worker.py

```diff
@@ -667,69 +662,68 @@ def _stop_job(
                     "deployment configuration."
                 )
 
-            current_cluster_uid = self._get_cluster_uid(client)
+            current_cluster_uid = await self._get_cluster_uid(client)
             if job_cluster_uid != current_cluster_uid:
                 raise InfrastructureNotAvailable(
                     f"Unable to kill job {job_name!r}: The job is running on another "
                     "cluster than the one specified by the infrastructure PID."
                 )
 
-            with self._get_batch_client(client) as batch_client:
-                try:
-                    batch_client.delete_namespaced_job(
-                        name=job_name,
-                        namespace=job_namespace,
-                        grace_period_seconds=grace_seconds,
-                        # Foreground propagation deletes dependent objects before deleting # noqa
-                        # owner objects. This ensures that the pods are cleaned up before # noqa
-                        # the job is marked as deleted.
-                        # See: https://kubernetes.io/docs/concepts/architecture/garbage-collection/#foreground-deletion # noqa
-                        propagation_policy="Foreground",
-                    )
-                except kubernetes.client.exceptions.ApiException as exc:
-                    if exc.status == 404:
-                        raise InfrastructureNotFound(
-                            f"Unable to kill job {job_name!r}: The job was not found."
-                        ) from exc
-                    else:
-                        raise
-
-    @contextmanager
-    def _get_configured_kubernetes_client(
+            batch_client = BatchV1Api(client)
```

Comment:
I tried it but doing so caused it to clean up resources early because the `close` function on each specific resource client just call the api clients `close`. I think it should be fine as the api client is already in a context manager of its own.